### ISSUE TITLE
ParaView with additional python-packages

### ISSUE THREAD

**Issue by jobojeha** [NixOS/nixpkgs#78438]
## Issue description
ParaView comes with it's own python interpreter `pvbatch` which can be used to run standard python scripts with additional ParaView functions.
In addition to the already provided numpy module I also need to import matplotlib (or every other python module really).

First I tried overrideAttrs on the [paraview package](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/graphics/paraview/default.nix):

```nix
{ config, pkgs, stdenv, ... }:

with pkgs;
  let
    pkgs-unstable = import <nixos-unstable> {};
    my-python-packages = python-packages: with python-packages; [
      numpy
      matplotlib
    ]; 
    python-with-my-packages = (python2.withPackages my-python-packages).override { 
      ignoreCollisions = true; 
    }; 
  in
{
home.packages = [ 
    (pkgs-unstable.paraview.overrideAttrs (oldAttrs: {

      buildInputs = oldAttrs.buildInputs ++ [ 
        python-with-my-packages
      ];
      postInstall = ''
        wrapProgram $out/bin/paraview \
          --set PYTHONPATH "${python-with-my-packages}/${pkgs.python.sitePackages}"
        wrapProgram $out/bin/pvbatch \
          --set PYTHONPATH "${python-with-my-packages}/${pkgs.python.sitePackages}"
        wrapProgram $out/bin/pvpython \
          --set PYTHONPATH "${python-with-my-packages}/${pkgs.python.sitePackages}"
          '';

    }))
  ];
}
```
 Although it works I don't want to recompile ParaView for every change in my python modules. So I read on the wiki that I could just wrap the binaries:

```nix
{ config, pkgs, ... }:

  let
    pkgs-unstable = import <nixos-unstable> {};
    pvbatchAdditionalPythonPackages = python-packages: with python-packages; [
      numpy
      matplotlib
    ]; 
    pvbatchPython = (pkgs.python2.withPackages pvbatchAdditionalPythonPackages).override { 
      ignoreCollisions = true; 
    }; 
  in
{
  


  
  home.packages = [ 
   
    (pkgs.runCommand "pvbatch" {
      buildInputs = [ pkgs.makeWrapper ];
      } ''
          mkdir $out
          ln -s ${pkgs-unstable.paraview}/* $out
          rm $out/bin
          mkdir $out/bin
          ln -s ${pkgs-unstable.paraview}/bin/* $out/bin
          rm $out/bin/pvbatch
          makeWrapper ${pkgs-unstable.paraview}/bin/pvbatch $out/bin/pvbatch \
            --set PYTHONPATH "${pvbatchPython}/${pkgs.python.sitePackages}"
        '')
  ];

}
````
But I guess because the binary is wrapped inside the builder **my additional wrapper is ignored**.

### Steps to reproduce
1. Build the posted home-config (I used home-manager)
2. run `pvbatch` with a python script calling `import matplotlib.pyplot`

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.4.0-18362-Microsoft, Ubuntu, 18.04.3 LTS (Bionic Beaver)`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.2`
 - channels(user): `"home-manager-19.09, nixos-unstable-20.03pre209840.a65f338ccbd, nixpkgs-20.03pre209992.a7a85a015c4"`
 - nixpkgs: `/home/user/.nix-defexpr/channels/nixpkgs`

Same goes for my NixOS system.

/cc @dguibert

**Comment by dguibert** [comment]
Maybe calling wrapWrapper with `--prefix` instead of `--set` in the paraview derivation.
A cleaner solution might be to implement `withPackages` in paraview likewise wthin python.

**Comment by dguibert** [comment]
solved by #78602
